### PR TITLE
CO-124: add repo stewardship audit

### PR DESCRIPTION
**Pull request by Kbediako**
## What

- Add `npm run repo:stewardship`, backed by a tracked-file audit script and stewardship catalog.
- Classify every tracked surface as `validate`, `update`, `delete`, or `retain_with_rationale`.
- Expand the weekly freshness workflow into a repo stewardship cadence that runs docs checks, docs freshness, and stewardship reporting.
- Refresh the CO-124 docs packet and post-merge March 11 freshness cohort so the current baseline stays green.

## Why

CO-124 asks for recurring repo-wide stewardship beyond the existing markdown-centric freshness surfaces. This keeps cleanup output reviewable and bounded while preserving CO-88, CO-75, and CO-102 ownership boundaries.

## Validation

- `node scripts/delegation-guard.mjs`
- `node scripts/spec-guard.mjs --dry-run`
- `npm run build`
- `npm run lint`
- `npm run test`
- `npm run docs:check`
- `npm run docs:freshness`
- `npm run repo:stewardship`
- `node scripts/diff-budget.mjs`
- forced `npm run review`: `status: succeeded`, `review_outcome: clean-success`
- explicit elegance/minimality pass
- `npm run pack:smoke`

Diff budget override: CO-124 is a docs-first stewardship lane that must land the audit script, tracked-file catalog, weekly workflow cadence, focused tests, and required task/docs registry metadata together so the recurring repo-wide contract is reviewable in one bounded PR.

Linear: CO-124

## Summary by CodeRabbit

* **New Features**
  * Adds an npm script for an automated repo-wide stewardship audit that inventories tracked files, classifies them (validate / update / delete / retain_with_rationale) and emits JSON/optional markdown reports; weekly CI now runs and publishes stewardship alongside docs freshness checks.

* **Documentation**
  * New PRD, tech spec, action plan, stewardship catalog and developer guidance; multiple docs, task specs and registries updated.

* **Tests**
  * New end-to-end and unit tests covering audit behaviours and edge cases.

### CODE REVIEW

File: AGENTS.md [hunk 1]
```diff
@@ -211,6 +211,7 @@ Implementation work is not “complete” until you run (in order):
 | `npm run test` | Unit + integration checks | Vitest harness covering orchestrator + patterns. |
 | `npm run docs:check` | Docs hygiene gate | Deterministically validates scripts/pipelines/paths referenced in agent-facing docs, current posture locks, bundled-skill roster parity, and the README front-door budget. |
 | `npm run docs:freshness` | Docs freshness gate | Validates registry coverage plus docs-catalog class coverage and emits a class-separated `out/<task-id>/docs-freshness.json`. |
+| `npm run repo:stewardship` | Repo stewardship audit | Inventories tracked files via `git ls-files`, classifies each surface as `validate`, `update`, `delete`, or `retain_with_rationale`, and emits `out/<task-id>/repo-stewardship.json`. |
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply AGENTS.md:215]: @coderabbitai Fixed in 795d1613f by adding `npm run repo:stewardship` to the mandatory ordered gate list and refreshing the AGENTS instruction stamp. Please re-review/resolve.

**Comment by Kbediako** [comment]
@codex review

File: scripts/lib/docs-helpers.js [hunk 2]
```diff
@@ -66,6 +67,28 @@ export async function collectDocFiles(repoRoot) {
...
 
+export function listTrackedFiles(repoRoot, pathspecs = []) {
+  const args = ['-C', repoRoot, 'ls-files'];
+  if (Array.isArray(pathspecs) && pathspecs.length > 0) {
+    args.push('--', ...pathspecs);
+  }
+
+  const git = spawnSync('git', args, {
+    encoding: 'utf8'
+  });
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply scripts/lib/docs-helpers.js:80]: @codex Fixed in 86be14453 by raising the `git ls-files` stdout buffer in `listTrackedFiles` to 64 MiB and adding focused coverage for the spawn options. Local validation passed through build, lint, full test, docs checks, repo stewardship, diff budget, pack smoke, standalone review, and elegance review. Please re-review/resolve.

File: scripts/lib/docs-helpers.js [hunk 3]
```diff
@@ -66,6 +68,29 @@ export async function collectDocFiles(repoRoot) {
...
+  }
+
+  const git = spawnSync('git', args, {
+    encoding: 'utf8',
+    maxBuffer: GIT_LS_FILES_MAX_BUFFER
+  });
+
+  if (git.status !== 0) {
+    throw new Error(`git ls-files failed while collecting tracked files: ${git.stderr || git.stdout}`);
+  }
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply scripts/lib/docs-helpers.js]: @coderabbitai Fixed in ab960c7b3 by including spawn errors, stderr/stdout, status, and signal in `git ls-files` failure diagnostics, plus focused coverage for pathspec forwarding, non-zero exits, and spawn ENOENT. Full local validation passed through pack:smoke; the bounded standalone review was executed and recorded a command-intent boundary fallback with manual correctness/elegance review. Please re-review/resolve.

File: scripts/lib/docs-helpers.js [hunk 4]
```diff
@@ -66,6 +68,38 @@ export async function collectDocFiles(repoRoot) {
...
+      `signal=${git.signal ?? '<no signal>'}`
+    ]
+      .filter((part) => part)
+      .join('; ');
+    throw new Error(`git ls-files failed while collecting tracked files: ${detail}`);
+  }
+
+  return git.stdout
+    .split('\n')
+    .map((line) => line.trim())
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply scripts/lib/docs-helpers.js]: @codex Fixed in d6d751c80 by switching `listTrackedFiles` to `git ls-files -z` and parsing NUL-delimited output without trimming path content, so valid leading/trailing spaces are preserved. Added focused whitespace-preservation coverage and reran full local validation through pack:smoke; bounded standalone review executed and recorded the expected command-intent fallback with manual correctness/elegance review. Please re-review/resolve.

File: scripts/repo-stewardship-audit.mjs [hunk 5]
```diff
@@ -0,0 +1,637 @@
...
+  --check                   Alias for default behavior
+  -h, --help                Show this help message`);
+}
+
+function normalizeString(value) {
+  return typeof value === 'string' ? value.trim() : '';
+}
+
+function normalizePath(value) {
+  const normalized = normalizeString(value).replace(/\\/g, '/');
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply scripts/repo-stewardship-audit.mjs]: Fixed in 1ac153f84. `normalizePath` now preserves exact path-string whitespace instead of trimming, and the audit regression tracks both ` leading.txt` and `trailing.txt ` through the real git inventory path. Full local gate batch is green through `pack:smoke` in `out/linear-d43b6785-88d6-442b-a34e-2ad19d4f723a/manual/20260411T015600Z-final-path-whitespace-validation/`.

**Comment by Kbediako** [comment]
@coderabbitai review

**Comment by Kbediako** [comment]
@coderabbitai resume

File: scripts/repo-stewardship-audit.mjs [hunk 6]
```diff
@@ -0,0 +1,637 @@
...
+        }
+      } else {
+        pattern += '[^/]*';
+      }
+      continue;
+    }
+
+    if (char === '?') {
+      pattern += '.';
+      continue;
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply scripts/repo-stewardship-audit.mjs:94]: @coderabbitai Fixed in 969dc14c0. `globToRegExp` now emits `[^/]` for `?`, and `tests/repo-stewardship-audit.spec.ts` verifies `docs/?/README.md` does not match `docs/a/b/README.md`. Full local validation is green through `pack:smoke` in `out/linear-d43b6785-88d6-442b-a34e-2ad19d4f723a/manual/20260411T023100Z-coderabbit-glob-readme-validation/`.

File: scripts/repo-stewardship-audit.mjs [hunk 7]
```diff
@@ -0,0 +1,637 @@
...
+    }
+
+    const parent = path.posix.dirname(currentDir);
+    if (!parent || parent === currentDir) {
+      break;
+    }
+    currentDir = parent;
+  }
+
+  return null;
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply scripts/repo-stewardship-audit.mjs:325]: @coderabbitai Fixed in 969dc14c0. README checks now preserve absent boundaries as `null`, treat explicit `readme_boundary: "."` as the repo root, and `findNearestLocalReadme` checks root `README.md` when the walk reaches `.`. Added regression coverage proving `scripts/tool.js` anchors to root `README.md`. Full local validation is green through `pack:smoke` in `out/linear-d43b6785-88d6-442b-a34e-2ad19d4f723a/manual/20260411T023100Z-coderabbit-glob-readme-validation/`.

File: scripts/repo-stewardship-audit.mjs [hunk 8]
```diff
@@ -0,0 +1,649 @@
...
+async function evaluateSurface(repoRoot, filePath, rule, context) {
+  const normalizedPath = normalizePath(filePath);
+  const label = getClassMeta(context.catalog, rule?.surface_class || 'uncatalogued').label;
+  let decision = rule?.decision ?? 'update';
+  const evidence = [];
+  const issues = [];
+  let readmeAnchor = null;
+  const surfaceExists = await pathExists(path.resolve(repoRoot, normalizedPath), {
+    allowMissingOnly: true
+  });
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply scripts/repo-stewardship-audit.mjs]: Fixed in 32ded1a65. Tracked surface presence now uses `lstat`, so a dangling tracked symlink is treated as the audited surface without requiring the target to exist; non-`ENOENT` errors still propagate. Added a regression that tracks a dangling symlink and verifies the audit keeps it as `validate`. Full local validation is green through `pack:smoke` in `out/linear-d43b6785-88d6-442b-a34e-2ad19d4f723a/manual/20260411T024500Z-codex-dangling-symlink-validation/`.

File: scripts/repo-stewardship-audit.mjs [hunk 9]
```diff
@@ -0,0 +1,659 @@
...
+  --check                   Alias for default behavior
+  -h, --help                Show this help message`);
+}
+
+function normalizeString(value) {
+  return typeof value === 'string' ? value.trim() : '';
+}
+
+function normalizePath(value) {
+  const normalized = typeof value === 'string' ? value.replace(/\\/g, '/') : '';
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply scripts/repo-stewardship-audit.mjs]: Fixed in 2c2e9b02a. normalizePath now preserves Git path text instead of rewriting literal backslashes, and the regression tracks a root file named foo\bar.txt through catalog matching and lstat-backed surface existence. Full local validation passed through pack:smoke in out/linear-d43b6785-88d6-442b-a34e-2ad19d4f723a/manual/20260411T025500Z-codex-backslash-path-validation/; standalone review completed with clean-success telemetry.
